custom_components/tesla_custom/number.py

```diff
@@ -0,0 +1,118 @@
+"""Support for the Tesla sensors."""
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity, NumberMode
+from homeassistant.core import HomeAssistant
+
+from . import TeslaDataUpdateCoordinator
+from .base import TeslaBaseEntity
+from .const import DOMAIN
+
+CHARGE_CURRENT_MIN = 1
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entities):
+    """Set up the Tesla Sensors by config_entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+    for car in hass.data[DOMAIN][config_entry.entry_id]["cars"]:
+        entities.append(TeslaChargeLimit(hass, car, coordinator))
+        entities.append(TeslaCurrentLimit(hass, car, coordinator))
+
+    async_add_entities(entities, True)
+
+
+class TeslaChargeLimit(TeslaBaseEntity, NumberEntity):
```

Comment:
Noting this here but `NumberEntity` seems to be unsupported and this needs to be updated.

````
2022-08-18 02:12:56.343 WARNING (MainThread) [homeassistant.components.number] Entity None (<class 'custom_components.tesla_custom.number.TeslaChargeLimit'>) is using deprecated NumberEntity features which will be unsupported from Home Assistant Core 2022.10, please report it to the custom integration author.
````